### PR TITLE
[Passkey #5] Auth startPasskeySignInWithCompletion: and finalizePasskeySignInWithPlatformCredential:completion

### PR DESCRIPTION
**Pull request by Xiaoshouzi-gh**
Added public APIs on FIRAuth level
- startPasskeySignInWithCompletion:
- finalizePasskeySignInWithPlatformCredential:completion:
- Added unit tests
- Added auth base64url utility methods to NSData.

TODO occurrences
1. finalizePasskeySignIn test cases will pass after we introduce the expires_in field.
2. Error codes will be added in a different CR.

### CODE REVIEW

File: FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuth.h [hunk 1]
```diff
@@ -578,6 +584,40 @@ NS_SWIFT_NAME(Auth)
                    completion:(nullable void (^)(FIRAuthDataResult *_Nullable authResult,
                                                  NSError *_Nullable error))completion;
 
+#if TARGET_OS_IOS || TARGET_OS_TV || TARGET_OS_OSX || TARGET_OS_MACCATALYST
+/**
+ @fn startPasskeySignInWithCompletion:
+ @brief start sign in with passkey retrieving challenge from GCIP and create an assertion request.
+ @param completion Optionally; a block which creates a assertation request.
+
+ @remarks //TODO add possible error codes
```
Comment (renkelvin) [inline FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuth.h:593]: What does this mean? Also add your ldap after TODO.

File: FirebaseAuth/Tests/Unit/FIRAuthTests.m [hunk 2]
```diff
@@ -1805,6 +1849,181 @@ - (void)testSignInAndRetrieveDataWithCustomTokenFailure {
...
+          XCTAssertEqual(error.code, FIRAuthErrorCodeOperationNotAllowed);
+          XCTAssertNotNil(error.userInfo[NSLocalizedDescriptionKey]);
+          [expectation fulfill];
+        }];
+    [self waitForExpectationsWithTimeout:kExpectationTimeout handler:nil];
+    OCMVerifyAll(_mockBackend);
+  }
+}
+
+// TODO @liubin uncomment this after expires_in changes
```
Comment (renkelvin) [inline FirebaseAuth/Tests/Unit/FIRAuthTests.m:1911]: We can include this in a future pr

Comment (Xiaoshouzi-gh) [reply FirebaseAuth/Tests/Unit/FIRAuthTests.m]: Got it, removed for now.  Will add it back with the "expires_in" fix PR.

File: FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuth.h (hunk 1 above)
Comment (Xiaoshouzi-gh) [reply FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuth.h]: Added ldap.

The reason for this is to follow the current public API reference format which includes a list of possible error codes.  Please see an example [here](https://github.com/firebase/firebase-ios-sdk/blob/passkey-user-change/FirebaseAuth/Sources/Public/FirebaseAuth/FIRAuth.h#L387).  I will fill this after error code bug (b/302700496) for backend is fixed.

**Review by renkelvin — APPROVED** [review]

**Pull request merged**